.changeset/stupid-peas-juggle.md

```diff
@@ -0,0 +1,10 @@
+---
+'astro': minor
+---
+
+Supports setting rootMargin for the `client:visible` directive.
```

Comment:
```suggestion
Extends the `client:visible` directive by adding an optional `rootMargin` property. This allows a component to be hydrated when it is close to the viewport instead of waiting for it to become visible.
```
Just suggesting something like the above, to provide a little more context here, given that this is a new feature. 

If you wanted to keep it short, then I'd go with:

> Adds an optional `rootMargin` property for the `client:visible` directive.

"Supports" isn't a word that indicates what *changed* as well as "Adds" does, especially when it's "support this new thing you probably don't know what it is/does". So in the interest of making these more helpful when skimmed, "Adds" is always a great choice for a new feature!

I would lean towards "Extends (or, Enhances, or Improves, if you prefer!) the `client:visible` directive" personally, to again make the change as clear for the reader as possible in terms of things they already understand: they know what this client directive is, and you've made it even better!